airflow/jobs/triggerer_job_runner.py

```diff
@@ -643,7 +647,9 @@ async def run_trigger(self, trigger_id, trigger):
             # allow triggers a chance to cleanup, either in that case or if
             # they exit cleanly. Exception from cleanup methods are ignored.
             with suppress(Exception):
-                await trigger.cleanup()
+                trigger_details = self.triggers[trigger_id]
+                if trigger.should_cleanup(trigger_details.get("termination_reason")):
```

Comment:
… and this should not use `get()` but just regular `[]`.

Reply:
Sure, already applied.